src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py

```diff
@@ -215,16 +220,36 @@ def get_case_description_attribute_loader_dict(
 
 def load_scalar(
     key: str, alfacase_content: DescriptionDocument, category: str
-) -> Scalar:
+) -> Scalar | ScalarExpression:
     """
-    Create a barril.units.Scalar instance from the given alfacase_content.
+    Create a barril.units.Scalar or ScalarExpression instance from the given alfacase_content.
     # TODO: ASIM-3556: All atributes from this module should get the category from the CaseDescription
     """
-    return Scalar(
-        category,
-        alfacase_content[key]["value"].content.data,
-        alfacase_content[key]["unit"].content.data,
-    )
+    value = alfacase_content[key]["value"].content.data
+    unit = alfacase_content[key]["unit"].content.data
+    match value:
+        case str():
+            return ScalarExpression(expr=value, unit=unit, category=category)
+        case int() | float():
+            return Scalar(category=category, value=value, unit=unit)
+        case unreachable:
+            assert_never(unreachable)
+
+
+def load_float(
+    key: str, alfacase_content: DescriptionDocument
+) -> float | FloatExpression:
+    """
+    Obtain the float data or create a FloatExpression.
+    """
+    value = alfacase_content[key].content.data
+    match value:
+        case str():
+            return FloatExpression(expr=value)
+        case float():
+            return value
+        case unreachable:
+            assert_never(unreachable)
```

Comment:
The same concerns about `Any` and that
```
        case unreachable:
            assert_never(unreachable)
```
and the not all floats will be "expressionable" (linkable?).